FormidableLabs/victory
# Allow safe user props to be passed to Candlestick, Errorbar, Histogram, Line, Pie, Polar-Axis, Scatter

### Description
Added functionality to allow user props that meet the safelist reqs to be passed to the following components: Candlestick, Errorbar, Histogram, Line, Pie, Polar-Axis, Scatter.

NOTE: User props will be filtered and only ones that match the user props safelist rules will be passed along to the component.

### Testing Procedures
Test the following examples for this PR:

- [Candlestick](http://localhost:3000/#/candlestick)
- [Error Bar](http://localhost:3000/#/errorbar)
- [Histogram](http://localhost:3000/#/histogram)
- [Line](http://localhost:3000/#/line)
- [Pie](http://localhost:3000/#/pie)
- [Polar-Axis](http://localhost:3000/#/axis)
- [Scatter](http://localhost:3000/#/scatter)

1. Inspect the first item on the example page with Dev Tools. This will be the element WITH a VictoryChart wrapper.  You should see a "data-test-variable" and an "aria-label" attribute attached to the `<g>` tag within the SVG.
2. Inspect the last item on the example page with Dev Tools. This will be the element WITHOUT a VictoryChart wrapper.  You should see a "data-test-variable" and an "aria-label" attribute attached to the `<svg>` tag within the VictoryContainer div.  A VictoryContainer gets added when the chart is considered standalone (not inside a VictoryChart wrapper).


NOTE: There are a few exceptions to where the attributes get added based on the way the components are created.  However, you should find them in the `<svg>` or `<g>` tags for that component when inspecting with DevTools. 

Exceptions to these 2 rules in this PR: 
- Histogram's and PolarAxis's attributes are on the svg tag for both standalone and wrapped Histograms.
- Pie charts are not typically wrapped in VictoryChart components. So the 2 examples are both without a wrapper. Also, both will show the attributes in the svg and g elements.
